jax/experimental/mosaic/gpu/examples/fdgg.py

```diff
@@ -0,0 +1,505 @@
+# Copyright 2025 The JAX Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Grouped GEMM kernel for Blackwell."""
+
+from dataclasses import dataclass
+import contextlib
+import itertools
+import math
+
+import jax
+from jax._src.interpreters import mlir
+from jax._src.lib.mlir import ir
+from jax._src.lib.mlir.dialects import arith
+from jax._src.lib.mlir.dialects import gpu
+from jax._src.lib.mlir.dialects import llvm
+from jax._src.lib.mlir.dialects import memref
+from jax._src.lib.mlir.dialects import nvvm
+from jax._src.lib.mlir.dialects import scf
+from jax.experimental.mosaic import gpu as mgpu
+from jax.experimental.mosaic.gpu import c, ds
+from jax.experimental.mosaic.gpu import tcgen05
+from jax.experimental.mosaic.gpu import mma_utils
+from jax.experimental.mosaic.gpu import profiler
+import jax.numpy as jnp
+import jax.random as jr
+import numpy as np
+from cuda.bindings.runtime import cudaDeviceGetAttribute, cudaDeviceAttr, cudaGetDeviceCount, cudaError_t
```

Comment:
I don't think we can depend on the CUDA Python APIs in those files. Is there some way to do what you want using the JAX device APIs? If not, we should consider exposing it